crates/matrix-sdk/src/oidc/mod.rs

```diff
@@ -202,19 +204,55 @@ use rand::{rngs::StdRng, Rng, SeedableRng};
 use ruma::api::client::discovery::discover_homeserver::AuthenticationServerInfo;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
-use tokio::sync::Mutex;
+use tokio::{spawn, sync::Mutex};
+use tracing::{trace, warn};
 use url::Url;
 
 mod auth_code_builder;
+mod cross_process;
 mod data_serde;
 mod end_session_builder;
 
+use self::cross_process::{
+    CrossProcessRefreshLockError, CrossProcessRefreshLockGuard, CrossProcessRefreshManager,
+};
 pub use self::{
     auth_code_builder::{OidcAuthCodeUrlBuilder, OidcAuthorizationData},
     end_session_builder::{OidcEndSessionData, OidcEndSessionUrlBuilder},
 };
 use crate::{authentication::AuthData, client::SessionChange, Client, RefreshTokenError, Result};
 
+type SaveSessionCallback =
+    dyn Send + Sync + Fn() -> Pin<Box<dyn Send + Sync + Future<Output = anyhow::Result<()>>>>;
+type ReloadSessionCallback = dyn Send + Sync + Fn() -> anyhow::Result<SessionTokens>;
```

Comment:
Having `Send` & `Sync` before the "main" trait of a `dyn` type seems unusual.

Reply:
Oh it's me being noobish-cautious and not knowing if the bound would apply to the result type of the function bound. But you're right, will fix.